lib/spack/spack/build_environment.py

```diff
@@ -511,6 +502,19 @@ def setup_package(pkg, dirty):
     validate(spack_env, tty.warn)
     spack_env.apply_modifications()
 
+    # All module loads that otherwise would belong in previous functions
+    # have to occur after previous line. They are now here
```

Comment:
IMO this should explicitly refer to the call to `spack_env.apply_modifications` and it may be work adding to the explanation like:

> spack_env includes modifications like unsetting LD_LIBRARY path, which required modules may update. Modules are therefore loaded after the call to spack_env.apply_modifications.